tests/memoryview/memoryview_annotation_typing.py

```diff
@@ -5,9 +5,23 @@
 from __future__ import annotations  # object[:] cannot be evaluated
 
 import cython
+try:
+    import typing
+except ImportError:
+    pass  # Workaround for python 2.7
 import numpy
 
 
+def slice_optional(m: typing.Optional[cython.double[:]]):
```

Comment:
In longer test files, I prefer a (loose) ordering from simple to more complex, start to end. So here, I'd rather keep the simple 1D and 2D tests at the beginning and only then add this test for `Optional[]` after them. Can go before the `nogil` tests or further down, I don't think that matters. But certainly after the trivial test cases.

Reply:
done